src/compositedataframe.jl

```diff
@@ -1,23 +1,93 @@
 using Compat
 
-export AbstractCompositeDataFrame, CompositeDataFrame
+export AbstractCompositeDataFrame, AbstractCompositeDataFrameRow, 
+       CompositeDataFrame, row
 
+"""
+    AbstractCompositeDataFrame
+    
+An abstract type that is an AbstractDataFrame. Each type that inherits from
+this is expected to be a type-stable data frame. 
+"""
 abstract AbstractCompositeDataFrame <: AbstractDataFrame
 
+abstract AbstractCompositeDataFrameRow
+
+
+"""
+    row(cdf::AbstractCompositeDataFrame, i)
+    
+Return the `CompositeDataFrameRow` object for row `i` of `cdf`. This object has
```

Comment:
I would have said "Return row `i` of `cdf` as a `CompositeDataFrameRow` object." Maybe more natural.